internal/pkg/agent/vault/vault_linux.go

```diff
@@ -59,42 +65,90 @@ func New(path string, opts ...OptionFunc) (v *Vault, err error) {
 		}
 	}
 
-	key, err := getOrCreateSeed(path, options.readonly)
+	r := &Vault{
+		path:       path,
+		retryDelay: options.retryDelay,
+		lock:       flock.New(filepath.Join(path, lockFile)),
+	}
+
+	err = r.tryLock(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = r.unlock(err)
+	}()
+
+	r.key, err = getOrCreateSeed(path, options.readonly)
 	if err != nil {
 		return nil, fmt.Errorf("could not get seed to create new valt: %w", err)
 	}
 
-	return &Vault{
-		path: path,
-		key:  key,
-	}, nil
+	return r, nil
+}
+
+// try to acquire exclusive lock
+func (v *Vault) tryLock(ctx context.Context) error {
+	_, err := v.lock.TryLockContext(ctx, v.retryDelay)
+	if err != nil {
+		return fmt.Errorf("failed to acquire exclusive lock: %v, err: %w", v.lock.Path(), err)
+	}
+	return err
```

Comment:
[Blocker]
Please, `err` is always `nil` so return `nil`.
```suggestion
	return nil
```

Reply:
done